autos.py

```diff
@@ -0,0 +1,51 @@
+
+from fastapi import Depends, HTTPException, APIRouter
+from sqlalchemy.orm import Session
+from database import SessionLocal, engine
+from typing import List, Optional
+import models
+import schemas
+import crud
+
+
+models.Base.metadata.create_all(bind=engine)
+
+router = APIRouter()
+
+
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@router.get("/autos", response_model=List[schemas.AutoModel])
+def read_autos(
+        db: Session = Depends(get_db),
+        color: Optional[str] = None,
+        ):
+    if color:
```

Comment:
just push all your optional params into paramsObject and give it to crud without IF